src/main/java/me/botsko/prism/utils/BlockUtils.java

```diff
@@ -27,586 +20,727 @@
+                return block.getRelative(BlockFace.DOWN);
+            }
+        }
+
+        return block;
+    }
+
+    /**
+     * True if flow will break it.
+     *
+     * @param m Material.
+     * @return bool
+     */
+    public static boolean canFlowBreakMaterial(Material m) {
+        return flowBreaks.isTagged(m);
+    }
+
+    /**
+     * True if requires Soil.
+     *
+     * @param m Material.
+     * @return bool
+     */
+    public static boolean materialRequiresSoil(Material m) {
+        return MaterialTag.CROPS.isTagged(m);
+    }
+
+    /**
+     * Fine connected blocks.
+     *
+     * @param type           Material
+     * @param currBlock      block
+     * @param foundLocations List
+     * @return List
+     */
+    public static ArrayList<Block> findConnectedBlocksOfType(Material type, Block currBlock,
+                                                             ArrayList<Location> foundLocations) {
+
+        ArrayList<Block> foundBlocks = new ArrayList<>();
+
+        if (foundLocations == null) {
+            foundLocations = new ArrayList<>();
+        }
+
+        foundLocations.add(currBlock.getLocation());
+
+        for (int x = -1; x <= 1; x++) {
+            for (int z = -1; z <= 1; z++) {
+                for (int y = -1; y <= 1; y++) {
+                    Block newBlock = currBlock.getRelative(x, y, z);
+                    // ensure it matches the type and wasn't already found
+                    if (newBlock.getType() == type && !foundLocations.contains(newBlock.getLocation())) {
+                        foundBlocks.add(newBlock);
+                        ArrayList<Block> additionalBlocks = findConnectedBlocksOfType(type, newBlock, foundLocations);
+                        if (additionalBlocks.size() > 0) {
+                            foundBlocks.addAll(additionalBlocks);
+                        }
+                    }
+                }
+            }
+        }
+
+        return foundBlocks;
+
+    }
+
+    /**
+     * Get Block below of same type.
+     *
+     * @param m   Material.
+     * @param loc Location
+     * @return BLock.
+     */
+    public static Block getFirstBlockOfMaterialBelow(Material m, Location loc) {
+        for (int y = (int) loc.getY(); y > 0; y--) {
+            loc.setY(y);
+            if (loc.getBlock().getType().equals(m)) {
+                return loc.getBlock();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * True if grows.
+     *
+     * @param m Material
+     * @return bool
+     */
+    public static boolean isGrowableStructure(Material m) {
+        return growableStructure.isTagged(m);
+    }
+
+    /**
+     * If same block id or core type.
+     *
+     * @param mat1 Material
+     * @param mat2 Material
+     * @return bool
+     */
+    public static boolean areBlockIdsSameCoreItem(Material mat1, Material mat2) {
+
+        // Get the obvious one out of the way.
+        if (mat1 == mat2) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)